ansible/roles/machine/provision/tasks/main.yml

```diff
@@ -6,7 +6,7 @@
       when: ansible_distribution == 'Fedora' and ansible_distribution_version is version('32', '<=')
 
     - name: get DNS server from resolvectl (fedora >= f33)
-      shell: "resolvectl status | grep 'Current DNS Server' | awk -F': ' '{print $2}'"
+      shell: "resolvectl dns | grep -m1 'Link' | awk -F': ' '{print $2}' | tr -d '[:space:]'"
```

Comment:
I tried this cmd on a 1minutetip machine but this would return a mangled output:
```
# resolvectl dns
Global:
Link 2 (eth0): 10.11.5.19 10.5.30.45
# resolvectl dns | grep -m1 'Link' | awk -F': ' '{print $2}' | tr -d '[:space:]'
10.11.5.1910.5.30.45
```
There are multiple issues:
- we can't assume that the DNS server is always configured at the link level, it could also be set for "Global: "
- multiple servers can be listed in each line

Something like `resolvectl dns | grep -m1 ': [1-9]' | awk -F': ' '{print $2}' | cut -d' ' -f 1` should work (discard lines without any IP address and consider only the first one, extract the content after ': ', split with space and take only the first one.

Reply:
Thanks @flo-renaud! I wasn't expecting the link to return more than one address.

What I don't understand about the this issue is why the [ipa-test-config.yaml](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/25a104c0-3c91-11eb-a77b-fa163ed58e5d/ipa-test-config.yaml) you linked shows 2 addresses (one from Global another from Link supposedly) because the one from Global should have been set by a task called later [`create /etc/resolv.conf file from template`](https://github.com/freeipa/freeipa-pr-ci/blob/77bb1540f8157558b65ec6a718162e8d1d3982d4/ansible/roles/machine/provision/tasks/main.yml#L92-L95). 

I failed to reproduce that.

The behaviour below shouldn't affect `ipa-test-config.yaml` generation, but If that happens like the above, then the global IP might be 'master's address, not runner's IP. 

These are the outputs from `resolve dns` in various stages:
* Before executing any playbook:
  ```
  $ resolvectl dns
  Global:
  Link 2 (eth0): 192.168.124.1
  ```
* After updating /etc/resolv.conf:
  ```
  resolvectl dns
  Global: 192.168.124.252
  Link 2 (eth0): 192.168.124.1
  ```
* Few minutes after executing that (because NM updated the `resolv.conf`)
  ```
  resolvectl dns
  Global: 127.0.0.1
  Link 2 (eth0): 192.168.124.1
  ```